noxfile.py

```diff
@@ -1,26 +1,44 @@
-"""A nox configuration file so that we can build the documentation easily with nox.
-- see the README.md for information about nox.
-- ref: https://nox.thea.codes/en/stable/
-"""
+import glob
+import os.path
+
 import nox
 
 nox.options.reuse_existing_virtualenvs = True
 
-build_command = ["-b", "dirhtml", "docs", "docs/_build/html"]
 
-@nox.session()
+@nox.session(python="3.13", default=False)
 def docs(session):
-    session.install("-r", "docs/requirements.txt")
-    if "live" in session.posargs:
-        # Add relative paths to this if we ever need to ignore them
-        AUTOBUILD_IGNORE = [
-          "docs/_data",
-          "docs/_build",
-        ]
+    """
+    Build the documentation.
+    """
+    sphinx_config_files = glob.glob("./docs/**/conf.py", recursive=True)
+    sphinx_config = sphinx_config_files[0]
+    source_dir = os.path.dirname(sphinx_config)
+    if source_dir.find("source") == -1:
+        output_dir = os.path.join(source_dir, "_build/html")
+    else:
+        output_dir = os.path.join(os.path.dirname(source_dir), "_build/html")
```

Comment:
I think this might be more complicated than it needs to be, it shouldn't need these conditionals and searching, you can set the values:

```python
source_dir = "docs/source"
output_dir = "docs/_build"
data_dir = "docs/_data"
autobuild_ignore = [output_dir, data_dir]
```

Reply:
I also think it might be simpler to have the simple strings and set them appropriately in each project. We aren't going to be keeping noxfiles in sync across repos, I imagine, and setting these values once per repo ~ever will probably be simpler and clearer than having code that tries to run everywhere. What do you think?